packages/gatsby/src/schema/node-model.js

```diff
@@ -447,7 +449,8 @@ class ContextualNodeModel {
   runQuery(args, pageDependencies) {
     return this.nodeModel.runQuery(
       args,
-      this._getFullDependencies(pageDependencies)
+      this._getFullDependencies(pageDependencies),
+      this.context.typedKeyValueIndexes
```

Comment:
I don't think this does anything, runQuery doesn't accept another argument.

Reply:
You can remove it from here and not accept it in context, IMO. NodeModel constructor is a better injection point.